api/db-cdr/generate-cdr/libproject/devstart.rb

```diff
@@ -59,6 +59,19 @@ def bq_ingest(tier, source_project, dataset_name, table_match_filter="", table_s
   # bq init --delete_credentials as recommended in the output.
   # TODO(RW-3768): Find a better solution for Google credentials in docker.
 
+  # validate the CDR's tier label against the user-supplied tier, as a safety check
+
+  cdr_metadata = JSON.parse(common.capture_stdout %{bq show --format=json #{source_fq_dataset}})
+  dataset_tier = cdr_metadata.dig('labels', 'data_tier')
+
+  if dataset_tier
+    unless dataset_tier == tier_name
+      raise ArgumentError.new("The dataset's access tier '#{dataset_tier}' differs from the requested '#{tier_name}'. Aborting.")
+    end
+  else
+    puts "WARNING: no data_tier label found for #{source_fq_dataset}"
```

Comment:
nit: consider `common.warning` instead

Reply:
Thanks.

"common" is a total mystery to me so it never occurs to me to use it.